From 1ea1f41f40ad183176dba0ace8b873443d6aa82d Mon Sep 17 00:00:00 2001
From: Krzysztof Gajdemski <songo@debian.org.pl>
Date: Thu, 21 Mar 2013 11:40:14 +0100
Subject: [PATCH 9/9] ipvsadm: Show 'ops' flag regardless of service
 persistence

In print_service_entry() 'ops' option is displayed only for services
which have the IP_VS_SVC_F_PERSISTENT flag set. However, it should be
displayed for any VS with one-packet scheduling because OPS can be
also used without persistence.

Signed-off-by: Krzysztof Gajdemski <songo@debian.org.pl>
Signed-off-by: Simon Horman <horms@verge.net.au>
---
 ipvsadm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ipvsadm.c b/ipvsadm.c
index 454287c..0197515 100644
--- a/ipvsadm.c
+++ b/ipvsadm.c
@@ -1517,9 +1517,9 @@ print_service_entry(ipvs_service_entry_t *se, unsigned int format)
 					printf(" mask %i", se->netmask);
 			if (se->pe_name[0])
 				printf(" pe %s", se->pe_name);
-			if (se->flags & IP_VS_SVC_F_ONEPACKET)
-				printf(" ops");
 		}
+		if (se->flags & IP_VS_SVC_F_ONEPACKET)
+			printf(" ops");
 	}
 	printf("\n");
 
-- 
1.8.1.4

